src/codewhisperer/util/authUtil.ts

```diff
@@ -91,6 +85,12 @@ export class AuthUtil {
         return this.secondaryAuth.activeConnection
     }
 
+    public get startUrl(): string | undefined {
+        // Reformat the url to remove any trailing '/' and `#`
+        // e.g. https://view.awsapps.com/start/# will become https://view.awsapps.com/start
+        return isSsoConnection(this.conn) ? this.reformatStartUrl(this.conn?.startUrl) : undefined
```

Comment:
Future followup: We need to stop adding special things to `codewhisperer/authUtil.ts`. Most auth-related code should live in the `shared/.../auth.ts`.